tests/suites/test_suite_x509write.data

```diff
@@ -3,15 +3,15 @@ depends_on:MBEDTLS_SHA1_C:MBEDTLS_RSA_C:MBEDTLS_PKCS1_V15
 x509_csr_check:"data_files/server1.key":"data_files/server1.req.sha1":MBEDTLS_MD_SHA1:0:0:0:0
 
```

Comment:
Please update dependencies in `test_suite_x509parse.data` as well.

Note: strictly speaking updating dependencies that don't break the CI is out of scope. However, it's easy to deal with them all, and you've already handled most of them. So I'm requesting to handle them all.

I used the following one-liner to look for problematic test cases:
```
perl -00 -ne '/sha.?(224|384)/i or next; $h=$1; /\ndepends_on:.*_SHA_?$h.*\n.*sha_?$h/i and next; print' tests/suites/test_suite_*.data
```
There may be a few false positive and false negatives, but it's close enough to be useful.